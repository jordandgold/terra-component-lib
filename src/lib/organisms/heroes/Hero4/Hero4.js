import React from "react";
import "./Hero4.scss";
import PropTypes from "prop-types";
import CTALinkSection from "../../../CTALinkSection/CTALinkSection";

const Hero4 = props => {
  const { title, text, ctas, video } = props;
  return (
    <section className="ter-hero-four">
      <div className="ter-hero-four__video-container">
        <iframe
          width="560"
          height="315"
          src={video.url}
          frameBorder="0"
          allow={`accelerometer; ${video.autoplay &&
            "autoplay"}; encrypted-media; gyroscope; picture-in-picture ${video.allowFullScreen &&
            "fullscreen"}`}
          className="ter-hero-four__video"
        />
      </div>
      <div className="ter-hero-four__content-container">
        <h2 className="ter-hero-four__title">{title}</h2>
        <p className="ter-hero-four__text">{text}</p>
        <CTALinkSection ctas={ctas} />
      </div>
    </section>
  );
};

export default Hero4;

Hero4.propTypes = {
  video: PropTypes.shape({
    url: PropTypes.string.isRequired,
    autoPlay: PropTypes.bool,
    allowFullScreen: PropTypes.bool
  }).isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  ctas: PropTypes.shape({
    ctaOne: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    ctaTwo: PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    }),
    subCTA: PropTypes.shape({
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired
    })
  })
};
