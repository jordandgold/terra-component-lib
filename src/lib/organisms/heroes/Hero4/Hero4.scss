@import "../../../global-styles/terra.scss";

.ter-hero-four {
  padding: spacing("lg");

  @media (min-width: $breakpoint-sm) {
    @include grid-container();
  }

  &__video-container {
    position: relative;
    width: 100%;
    height: 0;
    padding-bottom: 56.25%;

    @media (min-width: $breakpoint-sm) {
      @include grid-column(8);
    }
  }

  &__video {
    position: absolute;
    height: 100%;
    width: 100%;
  }

  &__content-container {
    @media (min-width: $breakpoint-sm) {
      @include grid-column(4);
    }

    @media (min-width: $breakpoint-md) {
      display: flex;
      flex-direction: column;
      justify-content: center;
    }
  }

  &__title {
    padding-top: spacing("md");

    @media (min-width: $breakpoint-sm) {
      padding-top: 0;
    }
  }

  &__text {
    margin-top: spacing("xs");
  }

  .ter-cta-section {
    @media (min-width: $breakpoint-xs) {
      width: 200px;
    }

    .ter-button {
      margin: spacing("xs") auto 0;
      text-align: center;
      display: block;

      @media (min-width: $breakpoint-xs) {
        margin: spacing("xs") 0 0;
      }
    }
  }

  .ter-cta-section__link {
    margin-top: spacing("xs");
    display: block;

    @media (min-width: $breakpoint-xs) {
      text-align: center;
    }
  }
}
