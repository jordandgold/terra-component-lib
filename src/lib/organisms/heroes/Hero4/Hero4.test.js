import React from "react";
import { shallow } from "enzyme";
import Hero4 from "./Hero4";

describe("Hero4", () => {
  let wrapper;
  const mockCTALinks = {
    ctaOne: {
      className: "ter-button--primary--1",
      text: "Button",
      link: "#"
    },
    ctaTwo: {
      className: "ter-button--secondary--1",
      text: "Button",
      link: "#"
    },
    subCTA: {
      text: "Button",
      link: "#"
    }
  };

  const mockVideo = { url: "test" };

  beforeEach(() => {
    wrapper = shallow(
      <Hero4
        title="Design is design."
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
        video={mockVideo}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot with video allowances", () => {
    const mockVideo = { url: "test", autoplay: true, allowFullScreen: true };

    wrapper = shallow(
      <Hero4
        title="Design is design."
        text="Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis."
        ctas={mockCTALinks}
        video={mockVideo}
      />
    );
  });
});
