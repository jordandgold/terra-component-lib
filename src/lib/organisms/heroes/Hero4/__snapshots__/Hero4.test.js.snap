// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero4 should match the snapshot 1`] = `
<section
  className="ter-hero-four"
>
  <div
    className="ter-hero-four__video-container"
  >
    <iframe
      allow="accelerometer; undefined; encrypted-media; gyroscope; picture-in-picture undefined"
      className="ter-hero-four__video"
      frameBorder="0"
      height="315"
      src="test"
      width="560"
    />
  </div>
  <div
    className="ter-hero-four__content-container"
  >
    <h2
      className="ter-hero-four__title"
    >
      Design is design.
    </h2>
    <p
      className="ter-hero-four__text"
    >
      Esse aliquip ad in et ut ipsum paEu elit consectetur aliquip excepteur fugiat ut qui dolor pariatur consectetur.riatur elit quis.
    </p>
    <CTALinkSection
      ctas={
        Object {
          "ctaOne": Object {
            "className": "ter-button--primary--1",
            "link": "#",
            "text": "Button",
          },
          "ctaTwo": Object {
            "className": "ter-button--secondary--1",
            "link": "#",
            "text": "Button",
          },
          "subCTA": Object {
            "link": "#",
            "text": "Button",
          },
        }
      }
    />
  </div>
</section>
`;
